File: list_lectures.js

const fs = require('fs');
const path = require('path');
const vm = require('vm');

const contentDir = path.join('e:', 'PPOM-UNO-Summary', 'content');

function listLectures() {
    const files = fs.readdirSync(contentDir).filter(f => f.endsWith('_content.js'));
    const lectures = [];

    for (const file of files) {
        const filePath = path.join(contentDir, file);
        const code = fs.readFileSync(filePath, 'utf8');

        // Clean code for VM
        let cleanCode = code.replace(/export\s+const\s+(\w+)/g, 'var $1');

        try {
            const sandbox = { window: {}, console: console, exports: {} };
            vm.createContext(sandbox);
            vm.runInContext(cleanCode, sandbox);

            // Find content
            const keys = Object.keys(sandbox).filter(k => k.endsWith('_CONTENT'));
            if (keys.length > 0) {
                const content = sandbox[keys[0]];
                let moduleName = "Unknown";

                // Try to extract module from title or file content if not explicit
                // Assuming `module` field might not exist in content object based on previous views, 
                // but checking `L2_content.js` showed `module: 'Neuro'`.
                if (content.module) moduleName = content.module;
                else if (content.title && content.title.includes(':')) moduleName = content.title.split(':')[0];

                lectures.push({
                    file: file,
                    id: content.id,
                    title: content.title || "No Title",
                    module: moduleName,
                    summaryHeader: content.summary ? content.summary.split('\n')[1] : ""
                });
            }
        } catch (err) {
            console.error(`Error processing ${file}: ${err.message}`);
        }
    }

    // Print csv-like
    console.log("File,ID,Module,Title");
    lectures.sort((a, b) => a.id.localeCompare(b.id, undefined, { numeric: true })).forEach(l => {
        console.log(`${l.file},${l.id},${l.module},"${l.title}"`);
    });
}

listLectures();

File: list_lectures.py

import os
import re

CONTENT_DIR = r"e:\PPOM-UNO-Summary\content"

def extract_metadata(content):
    id_match = re.search(r'id:\s*["\']([^"\']+)["\']', content)
    title_match = re.search(r'# Lecture.*?:(.*?)\n', content) # heuristic from summary markdown
    if not title_match:
         title_match = re.search(r'title:\s*["\']([^"\']+)["\']', content) # strictly from js object
         
    # Module is tricky, usually not in the JS object directly in previous files, but let's check.
    # L10 puts it in summary text: "**Session:** Lecture 10" or maybe inferred.
    # L2 had "module: 'Neuro'".
    module_match = re.search(r'module:\s*["\']([^"\']+)["\']', content)
    
    # If module is missing, guess from filename or title? 
    # The user said "ignore OMM and DPR lectures".
    # I can maybe detect OMM/DPR from the title or file text.
    
    is_omm_dpr = False
    if re.search(r'OMM|Osteopathic', content, re.IGNORECASE):
        is_omm_dpr = True
    if re.search(r'DPR|Doctoring', content, re.IGNORECASE): # DPR usually Doctoring/Patient...
        is_omm_dpr = True
        
    return {
        'id': id_match.group(1) if id_match else 'unknown',
        'title': title_match.group(1).strip() if title_match else 'Unknown',
        'module': module_match.group(1) if module_match else 'Unknown',
        'is_omm_dpr': is_omm_dpr
    }

def list_lectures():
    lectures = []
    for filename in os.listdir(CONTENT_DIR):
        if not filename.endswith('_content.js'):
            continue
            
        filepath = os.path.join(CONTENT_DIR, filename)
        with open(filepath, 'r', encoding='utf-8') as f:
            content = f.read()
            
        meta = extract_metadata(content)
        meta['filename'] = filename
        lectures.append(meta)
        
    # Sort
    lectures.sort(key=lambda x: x['id'])
    
    print(f"{'ID':<10} {'Module':<15} {'Excluded':<10} {'Title'}")
    print("-" * 80)
    for l in lectures:
        print(f"{l['id']:<10} {l['module']:<15} {str(l['is_omm_dpr']):<10} {l['title']}")

if __name__ == "__main__":
    list_lectures()
